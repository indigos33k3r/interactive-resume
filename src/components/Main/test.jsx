import React from 'react'
import {shallow} from 'enzyme'

describe('MainContainer', () => {
  it('renders <MainContainer />', () => {
    const component = render()

    expect(component).toMatchSnapshot()
  })

  context('when <CarPicker /> was clicked', () => {
    it('sets `state.playerCar` to passed value', () => {
      const component = render()
      component.find('CarPicker').prop('pickCarHandler')('batmobil')
      component.update()

      expect(component.state().playerCar).toBe('batmobil')
    })
  })

  function render() {
    const MainContainer = require('.').default

    return shallow(<MainContainer />)
  }
})
