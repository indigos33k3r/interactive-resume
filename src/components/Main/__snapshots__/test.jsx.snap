// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer renders <MainContainer /> 1`] = `
<section>
  <ContextProvider
    value={
      Object {
        "playerCar": null,
      }
    }
  >
    <CarPicker
      hide={false}
      pickCarHandler={[Function]}
    />
    <PageScroll />
    <PageContext />
  </ContextProvider>
</section>
`;
